src/solvers/flattening/arrays.cpp

```diff
@@ -273,14 +273,11 @@ void arrayst::add_array_constraints()
   }
 
   // add constraints for equalities
-  for(array_equalitiest::const_iterator it=
-      array_equalities.begin();
-      it!=array_equalities.end();
-      it++)
+  for(const auto & it : array_equalities)
```

Comment:
Please use a name different from `it` as this isn't an iterator anymore (and avoid the space after `&` for consistency)